staging/src/k8s.io/legacy-cloud-providers/vsphere/vclib/connection_test.go

```diff
@@ -187,12 +187,13 @@ func verifyWrappedX509UnkownAuthorityErr(t *testing.T, err error) {
 	if !ok {
 		t.Fatalf("Expected to receive an url.Error, got '%s' (%#v)", err.Error(), err)
 	}
-	x509Err, ok := urlErr.Err.(x509.UnknownAuthorityError)
+	verificationErr, ok := urlErr.Err.(*tls.CertificateVerificationError)
 	if !ok {
-		t.Fatalf("Expected to receive a wrapped x509.UnknownAuthorityError, got: '%s' (%#v)", urlErr.Error(), urlErr)
+		t.Fatalf("Expected to receive a wrapped *tls.CertificateVerificationError, got: '%s' (%#v)", urlErr.Err.Error(), urlErr.Err)
 	}
-	if msg := x509Err.Error(); msg != "x509: certificate signed by unknown authority" {
-		t.Fatalf("Expected 'signed by unknown authority' error, got: '%s'", msg)
+	_, ok = verificationErr.Err.(x509.UnknownAuthorityError)
+	if !ok {
+		t.Fatalf("Expected to receive a wrapped x509.UnknownAuthorityError, got: '%s' (%#v)", verificationErr.Err.Error(), verificationErr.Err)
 	}
```

Comment:
This commit gets rid of the check for the text of the x509.CertificateVerificationError error since we already check if the erorr is that type.

Reply:
this should be written in a way that works with both go1.20 and go1.19... would something along these lines work with both?

```
var x509Err x509.UnknownAuthorityError
if !errors.As(urlErr.Err, &x509Err) {
  fatal
}
...
```

Reply:
Oh that's a great idea